### ISSUE TITLE
Suggested change for `use-implicit-booleaness-not-comparison` doesn't seem right

### ISSUE THREAD

**Issue by clavedeluna** [pylint-dev/pylint#9353]
### Bug description

```python
z = []
res = z != []
```

### Configuration

```ini
enable=use-implicit-booleaness-not-comparison
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
test.py:2:6: C1803: "z != []" can be simplified to "z", if it is strictly a sequence, as an empty list is falsey (use-implicit-booleaness-not-comparison)
```

### Expected behavior

While in an if statement `if z != []: pass` The suggestion is right, in an assignment context I don't think it is. Changing
`res = z != []` to `res = z` is not the same thing

### Pylint version

```shell
pylint 3.0.0
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. ``bool(z)`` should be the suggestion when the result is not in a boolean context where the bool cast is implicit (for example in ``if z``)
